### ISSUE TITLE
Does not work with SSR yet

### ISSUE THREAD

**Issue by ntaloventi** [table-library/react-table-library#9]
`import { Table } from '@table-library/react-table-library/table'`

showing error

`Server Error
SyntaxError: Unexpected token 'export'
This error happened while generating the page. Any console logs will be displayed in the terminal window.`

**Comment by rwieruch** [comment]
Oh, that's strange. Does it disappear when removing the import?

**Comment by ntaloventi** [comment]
yup, it dissapear when remove the import statement

here is my package.json

{
  "name": "mylte",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "@material-ui/core": "^4.11.4",
    "@material-ui/lab": "^4.0.0-alpha.58",
    "axios": "^0.21.1",
    "js-cookie": "^2.2.1",
    "material-ui-snackbar-provider": "^1.4.0",
    "next": "10.2.2",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-hyper-tree": "^0.2.18",
    "react-table": "^7.7.0"
  }
}

**Comment by rwieruch** [comment]
It seems like NextJs does not use ES6 libraries, only libraries that are compiled to ES5. As workaround you can do something like this in NextJs:

```
const withTM = require("next-transpile-modules")([
  "@table-library/react-table-library",
]);
const withPlugins = require("next-compose-plugins");

module.exports = withPlugins([withTM], {});
```

However, it's still not compatible with SSR afterward, because this library uses Styled Components which apparently uses `window`. I need to figure out how to get rid of it.

**Comment by rwieruch** [comment]
Fixed here https://github.com/table-library/react-table-library/commit/27c2c6d26630e29744c55b765e80cd19b371036c and works with SSR now.

**Comment by darrenbridenbeck-measurabl** [comment]
Thanks for all your stellar work maintaining this project! 

Is `<Table />` still SSR safe though? I'm trying to use this library in a stripped down nextjs app and can't get it to render on the server. My understanding is that [the truthiness check for](https://github.com/table-library/react-table-library/blob/3b0ddab82cf7c633731087ba9534020ec9099db2/src/table/Table/index.tsx#L123) `calledOnce` before rendering the table is what's causing the issue.

tl;dr - refs only get a value when they're rendered as a DOM element, which can only happen on the client, [saeyth the react docs](https://react.dev/learn/referencing-values-with-refs#refs-and-the-dom):
> When you pass a ref to a ref attribute in JSX, like `<div ref={myRef}>`, React will put the corresponding DOM element into myRef.current
